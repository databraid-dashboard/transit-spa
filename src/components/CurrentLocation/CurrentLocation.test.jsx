import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
  CurrentLocation,
  mapDispatchToProps,
  mapStateToProps } from './CurrentLocation';

describe('CurrentLocation', () => {
  it('should render Determining your current location...', () => {
    const component = shallow(<CurrentLocation />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('calls updateCurrentLocation on load', () => {
    const updateCurrentLocation = jest.fn();
    mount(<CurrentLocation />);

    expect(updateCurrentLocation).toHaveBeenCalled();
  });

  it('mapStateToProps should return geolocating flag and current location', () => {
    const state = { configuration: {
      geolocating: true,
      currentLocation: {
        address: '44 Tehama St, San Francisco, CA 94105',
        lat: 37.7873889,
        lng: -122.3964106 },
    } };

    const expected = { geolocating: true,
      currentLocation: {
        address: '44 Tehama St, San Francisco, CA 94105',
        lat: 37.7873889,
        lng: -122.3964106 },
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });

  it('mapDispatchToProps should have property updateCurrentLocation', () => {
    const dispatch = jest.fn();

    expect(mapDispatchToProps(dispatch)).toHaveProperty('updateCurrentLocation');
  });
});
