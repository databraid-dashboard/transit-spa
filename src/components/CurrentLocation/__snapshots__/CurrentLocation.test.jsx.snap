// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentLocation should render Determining your current location... 1`] = `
<Segment>
  <h1>
    Determining your current location...
  </h1>
</Segment>
`;
